lib/client/okta.go

```diff
@@ -41,9 +40,19 @@ type OktaClient struct {
 	SessionToken string
 	Expiration   time.Time
 	BaseURL      *url.URL
-	MFAConfig    MFAConfig
-	Keyring      *keyring.Keyring
+	sessions     SessionCache
 	client       http.Client
+	selector     MFAInputs
+}
+
+type MFAInputs interface {
+	ChooseFactor(factors []MFAConfig) (int, error)
+	CodeSupplier(factor MFAConfig) (string, error)
+}
+
+type SessionCache interface {
```

Comment:
Ah yes, much better. 

I think we could shift some of the logic from here into the session cache itself. For example, knowing/generating the `key` I think should be the responsibility of the session cache, not the caller.

The `getSessionCookieKeyring` could move there as well.

Also, `data` is always just the session cookie, right? Might as well name it that way

The interface would then be something like

```
type SessionCache struct {
	Get(username string, domain string) ([]byte, error)
	Put(username string, domain string, cookie []byte) error
}
```

I wonder if `Get` and `Put` will ever need more info than the username and domain? If that's the case, you might want to pass in a struct that we can expand, to avoid having to change this API.

While writing this, I realized I'm basically just describing the other `sessioncache` :P